### PR TITLE
chore: Add typesVersions to package.json

### PR DESCRIPTION
**Pull request by benlesh**
This is an attempt to fix issues with VS Code auto-import, as described here in this issue: https://github.com/microsoft/TypeScript/issues/43034\#issuecomment-820621280

Related #6067

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,6 +6,9 @@
   "module": "./dist/esm5/index.js",
   "es2015": "./dist/esm/index.js",
   "types": "./dist/types/index.d.ts",
+  "typesVersions": {
+    ">=4.0": { "*": [ "dist/types/*" ] }
```
Comment (cartant) [inline package.json:10]: question: Could we use this to express that the minimum supported version is 4.2? I mean, we're not going to support anything older than that.

**Review by cartant — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (benlesh) [reply package.json:10]: Probably?

**Comment by benlesh** [comment]
This didn't appear to work at all. We can probably revert it. :\

**Comment by benlesh** [comment]
Shit. I wanted to get this into the RC :\

**Pull request merged**